_studio/mfx_lib/shared/src/libmfxsw_encode.cpp

```diff
@@ -269,6 +425,13 @@ VideoENCODE* _mfxSession::Create<VideoENCODE>(mfxVideoParam& par)
     {
         return nullptr;
     }
+    // if ctorFallback present, create FallbackProxyENCODE
+    if (handlers->second.fallback.ctor)
+    {
+        return new FallbackProxyENCODE(core, &m_coreInt, CodecId, MFX_PLATFORM_HARDWARE == m_currentPlatform, fei);
```

Comment:
That's over complex. Just use the handle you found to initialize FallbackProxyENCODE: `new FallbackProxyENCODE(handlers)` You will avoid second search in Fallback class and use codec definition from the handlers table right away. I.e. store reference to the handler in fallback class.